src/vm/eventpipeblock.cpp

```diff
@@ -57,69 +100,322 @@ bool EventPipeBlock::WriteEvent(EventPipeEventInstance &instance)
 
     if (m_pBlock == NULL)
     {
-        return false;
+        return;
     }
 
-    unsigned int totalSize = instance.GetAlignedTotalSize();
-    if (m_pWritePointer + totalSize >= m_pEndOfTheBuffer)
+    _ASSERTE(m_pWritePointer <= m_pEndOfTheBuffer);
+
+    memset(m_pBlock, 0, GetSize());
+    m_pWritePointer = m_pBlock;
+}
+
+EventPipeEventBlockBase::EventPipeEventBlockBase(unsigned int maxBlockSize, EventPipeSerializationFormat format, bool fUseHeaderCompression) :
+    EventPipeBlock(maxBlockSize, format), m_fUseHeaderCompression(fUseHeaderCompression)
+{
+    memset(m_compressedHeader, 0, 100);
+    Clear();
+}
+
+void EventPipeEventBlockBase::Clear()
+{
+    EventPipeBlock::Clear();
+    m_lastHeader.MetadataId = 0;
+    m_lastHeader.SequenceNumber = 0;
+    m_lastHeader.ThreadId = 0;
+    m_lastHeader.CaptureThreadId = 0;
+    m_lastHeader.StackId = 0;
+    m_lastHeader.TimeStamp.QuadPart = 0;
+    m_lastHeader.ActivityId = { 0 };
+    m_lastHeader.RelatedActivityId = { 0 };
+    m_lastHeader.DataLength = 0;
+
+    m_minTimeStamp.QuadPart = LLONG_MAX;
+    m_maxTimeStamp.QuadPart = LLONG_MIN;
+}
+
+void WriteVarUInt32(BYTE* & pWritePointer, unsigned int value)
```

Comment:
nit: Can we make `WriteVarUInt32` and `WriteVarUInt64` a single template function?